projects/packages/ip/src/class-utils.php

```diff
@@ -179,31 +211,339 @@ public static function get_ip_addresses_from_string( $ips ) {
 		return $result;
 	}
 
+	/**
+	 * Validates CIDR notation for IPv4 and IPv6 addresses.
+	 *
+	 * @param string $cidr CIDR notation IP address.
+	 * @return bool True if valid, false otherwise.
+	 */
+	public static function validate_cidr( $cidr ) {
+		// Split the CIDR notation into IP address and prefix length using the '/' separator.
+		$parts = explode( '/', $cidr );
+		if ( count( $parts ) !== 2 ) {
+			return false; // Invalid CIDR notation if it doesn't contain exactly one '/'.
+		}
+
+		list( $ip, $prefix ) = $parts;
+
+		// Validate the IP address.
+		if ( ! filter_var( $ip, FILTER_VALIDATE_IP ) ) {
+			return false;
+		}
+
+		$ip_version = self::get_ip_version( $ip );
+
+		// Validate prefix length.
+		if ( $ip_version === 'ipv4' ) {
+			// For IPv4, the prefix length must be an integer between 0 and 32.
+			if ( ! ctype_digit( $prefix ) || $prefix < 0 || $prefix > 32 ) {
+				return false;
+			}
+		} elseif ( $ip_version === 'ipv6' ) {
+			// For IPv6, the prefix length must be an integer between 0 and 128.
+			if ( ! ctype_digit( $prefix ) || $prefix < 0 || $prefix > 128 ) {
+				return false;
+			}
```

Comment:
I've gone ahead and used `self::validate_netmask()` here to avoid the code duplication! :+1:

I do kind of like enforcing the numeric netmask up front in the parsing step, so that the parsing function deals with extracting it from the string and ensuring it is cast to the correct type right away, and the validation function only needs to validate the value of the netmask (no passing in strings to validate, that we will always want to use as integers anyways, if the validation passes).

No opinions strongly held though!